src/main/java/com/tsmms/skoop/security/CurrentUserService.java

```diff
@@ -1,38 +1,44 @@
 package com.tsmms.skoop.security;
 
-import com.tsmms.skoop.exception.NoSuchResourceException;
 import com.tsmms.skoop.exception.UserNotAuthenticatedException;
-import com.tsmms.skoop.user.User;
-import com.tsmms.skoop.user.query.UserQueryService;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.oauth2.jwt.Jwt;
 import org.springframework.stereotype.Service;
 
-import static com.tsmms.skoop.exception.enums.Model.USER;
 import static com.tsmms.skoop.security.JwtClaims.SKOOP_USER_ID;
 
 @Service
 public class CurrentUserService {
 
-	private final UserQueryService userQueryService;
-
-	public CurrentUserService(UserQueryService userQueryService) {
-		this.userQueryService = userQueryService;
-	}
-
-	public User getCurrentUser() {
+	public String getCurrentUserId() {
 		if (SecurityContextHolder.getContext().getAuthentication() == null) {
 			throw new UserNotAuthenticatedException();
 		}
 
 		if (!(SecurityContextHolder.getContext().getAuthentication().getPrincipal() instanceof Jwt)) {
 			throw new UserNotAuthenticatedException();
 		}
-		final String userId = ((Jwt) SecurityContextHolder.getContext().getAuthentication().getPrincipal()).getClaimAsString(SKOOP_USER_ID);
-		return userQueryService.getUserById(userId).orElseThrow(() -> NoSuchResourceException.builder()
-				.model(USER)
-				.searchParamsMap(new String[]{"id", userId})
-				.build());
+		return  ((Jwt) SecurityContextHolder.getContext().getAuthentication().getPrincipal()).getClaimAsString(SKOOP_USER_ID);
+	}
+
+	/**
+	 * Checks whether the given user ID equals the user ID of the authenticated user.
+	 *
+	 * @param userId User ID to check against the authenticated user.
+	 * @return <code>true</code> if the given user ID equals the user ID of the authenticated user.
+	 */
+	public boolean isAuthenticatedUserId(String userId) {
+		return isAuthenticatedUserId(getCurrentUserId(), userId);
+	}
+	/**
+	 * Checks whether the given user ID equals the user ID of the authenticated user.
+	 *
+	 * @param authenticatedUserId User ID of the authenticated user.
+	 * @param userId User ID to check against the authenticated user.
+	 * @return <code>true</code> if the given user ID equals the user ID of the authenticated user.
+	 */
+	public boolean isAuthenticatedUserId(String authenticatedUserId, String userId) {
```

Comment:
Do we really need this as a public method? It does not seem to be used anywhere in the production code. The comparison of the user IDs could be embedded into the method above.

Reply:
No, we do not need it. The method was used before and I left it as it might have been used in future. But I agree with you in sense that as far as we do not need the method we can remove it.
So I removed the method and placed comparison of the user IDs in the method above.